features/step_definitions/formatter_steps.ts

```diff
@@ -31,3 +31,10 @@ Then(
     expect(actual).to.eql(expected)
   }
 )
+
+Then('the html formatter output is complete', async function (this: World) {
+  const actualPath = path.join(this.tmpDir, `html.out`)
+  const actual = await fs.readFile(actualPath, 'utf8')
+  expect(actual).to.contain('<html lang="en">')
+  expect(actual).to.contain('</html>')
```

Comment:
Really this is just testing that the formatter wrote out a page and (probably) got to the end, which is a bit more than we were doing before.

Maybe this is a bit overkill, but we could add a screenshot-based test that ensures e.g. the header part of the report has rendered as expected - this would help us catch issues manifesting as a JS error in the output page. What do you think?

Reply:
> Really this is just testing that the formatter wrote out a page and (probably) got to the end, which is a bit more than we were doing before.
> 

I think that's a good enough test.

> Maybe this is a bit overkill, but we could add a screenshot-based test that ensures e.g. the header part of the report has rendered as expected - this would help us catch issues manifesting as a JS error in the output page. What do you think?

If we're going to eye-ball it ("manually test it") we might as well just open the HTML in a browser and eyeball it that way. If there are any runtime errors we could inspect them further. Something you can't do in a png image.

I have nothing against manual testing. This is a perfect example of where it's the best way to test.

Reply:
Very fair